tests/test_utils.py

```diff
@@ -137,3 +143,87 @@ def jy2ksr_nonastropy(freq_arr):
     conv1 = jy2ksr_nonastropy(freqs) * units.K * units.sr / units.Jy
 
     assert np.allclose(conv0, conv1)
+
+
+@pytest.mark.parametrize(
+    ("fspec", "use_cli"), [("freqs", True), ("freqs", False), ("redshifts", False)]
+)
+def test_flat_spectrum_skymodel(fspec, use_cli, tmp_path, capsys):
+    n_freq = 20
+    freqs = np.linspace(150e6, 180e6, n_freq)
+    nside = 256
+    variance = 1e-6
+
+    redshifts = skyutils.f21 / freqs - 1
+    z_order = np.argsort(redshifts)
+    redshifts = redshifts[z_order]
+
+    if use_cli:
+        # cli only accepts frequencies
+        file_name = str(tmp_path) + "test_flat_spectrum.skyh5"
+        cli.make_flat_spectrum_eor(
+            [
+                "-v",
+                str(variance),
+                "--nside",
+                str(nside),
+                "--filename",
+                file_name,
+                "-s",
+                str(freqs[0]),
+                "-e",
+                str(freqs[-1]),
+                "-N",
+                str(n_freq),
+            ]
+        )
+        captured = capsys.readouterr()
+        assert captured.out.startswith(
+            "Generating sky model, nside 256, and variance 1e-06 K^2 at channel 0.\n"
+            "Generated flat-spectrum model, with spectral amplitude 0.037 K$^2$ Mpc$^3$"
+        )
+        sky = SkyModel.from_file(file_name)
+    else:
+        fspec_kwargs = {}
+        if fspec == "freqs":
+            fspec_kwargs = {"freqs": freqs * units.Hz}
+        else:
+            fspec_kwargs = {"redshifts": redshifts, "ref_zbin": n_freq - 1}
+
+        sky = skyutils.flat_spectrum_skymodel(
+            variance=variance, nside=nside, **fspec_kwargs
+        )
+
+    npix = 12 * nside**2
+    assert sky.Ncomponents == npix
+    np.testing.assert_allclose(sky.freq_array.value, freqs)
+    calc_var = np.var(sky.stokes, axis=2)
+
+    dz = redshifts[-1] - redshifts[-2]
+    omega = 4 * np.pi / npix
+    vol = (
+        Planck15.differential_comoving_volume(redshifts[-1]).to_value("Mpc^3/sr")
+        * dz
+        * omega
+    )
+
+    pspec_amp = variance * vol
+    assert f"{pspec_amp:.3f} " + r"K$^2$ Mpc$^3$" in sky.history
+
+    assert np.isclose(calc_var[0, 0].value, variance)
```

Comment:
I think these are pretty good tests, especially the direct comparison of the variance with the input.  I don't know if there's much else to do, there's no trivial way to get P(k) from a set of HEALPix maps.  I guess you could look at C_ell, but I don't know if that's really too relevant.